lib/openstax/aws/stack.rb

```diff
@@ -107,6 +176,29 @@ def wait_for_deletion
                            word: "deleted") if !dry_run
     end
 
+    def resource(logical_id)
+      stack_resource = aws_stack.resource(logical_id)
+
+      case stack_resource.resource_type
+      when "AWS::AutoScaling::AutoScalingGroup"
```

Comment:
this is a simple conditional, consider if over case (if has much better performance i think, and seems like it'd be easier to read)

Reply:
This case statement will grow to include options for lots of other resource types (e.g. buckets, IAM groups, etc, etc).